src/openms/source/KERNEL/MassTrace.cpp

```diff
@@ -294,15 +315,48 @@ namespace OpenMS
     return t_area;
   }
 
+  double MassTrace::computeMedianIntensity_() const
+  {
+    // determine median of intensities
+    double t_area(0);
+    std::vector<double> ints;
+    ints.reserve(trace_peaks_.size());
+    for (Size i=0; i<trace_peaks_.size(); ++i)
+    {
+      ints.push_back(trace_peaks_[i].getIntensity());
+    }
+    sort(ints.begin(), ints.end());
+    t_area = (ints.size() % 2 == 0 ? 0.5*(ints[ints.size()/2-1]+ints[ints.size()/2]) 
+      : ints[ints.size()/2]);
+    return t_area;
+  }
 
   double MassTrace::getIntensity(bool smoothed) const
   {
     if (smoothed)
-    {
-      return computeFwhmAreaSmooth();
+    { // will be removed soon
+      switch (quant_method_)
+      {
+        case MT_QUANT_AREA:
+          return computeFwhmAreaSmooth();
+        default:
+          throw Exception::InvalidValue(__FILE__, __LINE__, __PRETTY_FUNCTION__, "Member 'quant_method_' has unsupported value.", String(quant_method_));
```

Comment:
maybe add here to the exception that MT_QUANT_MEDIAN does not work when smoothed?